### PR TITLE
feat: initialize instrument detail page

### PR DESCRIPTION
**Pull request by kunfang98927**
1. **Added CSS for instrument detail view**:
    - Created `detail.css` to place most of styles of instrument detail page;
    - Updated `index.css` for instrument module global style;

2. **Added JavaScript for instrument detail interactions**:
    - Created `InstrumentDetail.js`: initialize `edit`, `cancel` and `publish` button listeners;

3. **Created and updated HTML templates**:
    - Created `detail.html`: Initialize instrument detail html #32 ;
    - Updated `masonryView.html` and `stdView.html`: when clicking on an image;

4. **Added view for instrument detail**:
    - Created `instrument_detail.py`: initialize the `InstrumentDetail` class as a `DetailView` #33 ;
    - Improved query efficiency by utilizing `context["instrument"]` instead of `get_object`.
    - Prepared for future language support by retrieving `active_language` for the detail page.

5. **Updated URL patterns to include instrument detail page**:
    - Modified `urls.py`: add `InstrumentDetail` view;

Resolves: #32
Resolves: #33

### CODE REVIEW

File: web-app/django/VIM/apps/instruments/templates/instruments/detail.html [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                  <div class="button-group">
+                    <button class="btn edit">Edit</button>
+                    <button class="btn cancel">Cancel</button>
+                    <button class="btn publish">Publish</button>
+                  </div>
+                </div>
+              </td>
+            </tr>
+            <tr>
+              <th scope="row">Classification</th>
```
Comment (dchiller) [inline web-app/django/VIM/apps/instruments/templates/instruments/detail.html:57]: I think this should probably be "MIMO Classification" or "Musical Instrument Museums Online Classification"

File: web-app/django/VIM/apps/instruments/views/instrument_detail.py
Comment (dchiller) [inline web-app/django/VIM/apps/instruments/views/instrument_detail.py:1]: Making a few comments on this file in places where the code as you have written increases the number of SQL queries more than is necessary. It's honestly fine for now, but could lead to much more overhead when we have many more than two languages.

File: web-app/django/VIM/apps/instruments/views/instrument_detail.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    Displays details of a specific instrument.
+    """
+
+    model = Instrument
+    template_name = "instruments/detail.html"
+    context_object_name = "instrument"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        instrument = self.get_object()
```
Comment (dchiller) [inline web-app/django/VIM/apps/instruments/views/instrument_detail.py:16]: `context["instrument"]` will already have the `instrument` you want, so you don't need to `get_object` (this leads to a duplicate query).

File: web-app/django/VIM/apps/instruments/views/instrument_detail.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        context["default_image_url"] = (
+            instrument.default_image.url if instrument.default_image else None
+        )
+        context["hornbostel_sachs_class"] = instrument.hornbostel_sachs_class
+        context["mimo_class"] = instrument.mimo_class
+
+        # Get english name
+        context["instrument_name_en"] = instrument.instrumentname_set.get(
+            language__en_label="english"
+        ).name
```
Comment (dchiller) [inline web-app/django/VIM/apps/instruments/views/instrument_detail.py:28]: You'll get the english instrument name below when you loop through all the names and you could extract it into a separate context item there, so again, this leads to 1 extra SQL query.

File: web-app/django/VIM/apps/instruments/views/instrument_detail.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+from django.views.generic import DetailView
+from VIM.apps.instruments.models import Instrument, Language
```
Comment (dchiller) [inline web-app/django/VIM/apps/instruments/views/instrument_detail.py:2]: `Lanugage` import is not used...are you using pylint for linting?

Also, I see now that this doesn't have the right dependencies for typing -- I'll fix this.

File: web-app/django/VIM/apps/instruments/views/instrument_detail.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            instrument_names.append(
+                {
+                    "name": instrument_name.name,
+                    "source_name": instrument_name.source_name,
+                    "language_code": instrument_name.language.wikidata_code,
+                    "language_id": instrument_name.language.wikidata_id,
+                    "language_en_label": instrument_name.language.en_label,
+                    "language_autonym": instrument_name.language.autonym,
+                }
+            )
```
Comment (dchiller) [inline web-app/django/VIM/apps/instruments/views/instrument_detail.py:42]: Ok, this is the place where I really do think you'd want to go ahead and optimize. Every step in the loop, this means you'll have a separate SQL query, which means if we end up having a hundred names for the instrument, we'll have a hundred queries. 

On line 32, we'll do one query to get all of the instrument names for the instrument, and then starting on line 37, we'll do another query to get the language info for each name.

If you use the [`.select_related` method](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#select-related), you can get this all in one query (essentially, you are telling Django to make a query with a join):
`instrument.instrumentname_set.all().select_related("language")`

**Comment by dchiller** [comment]
@kunfang98927 The page looks great... the only thing is that clicking the "Edit" buttons doesn't do anything for me. Should they? Or are they just placeholders for now?

**Comment by dchiller** [comment]
Also, are you using the Django Debug Toolbar and the django debugger from the Django extensions module? They are two of the development dependencies and can be very useful.

The toolbar, for example, can help identify where you are making extra queries... let me know if you have questions or want help setting them up :)

**Comment by kunfang98927** [comment]
> @kunfang98927 The page looks great... the only thing is that clicking the "Edit" buttons doesn't do anything for me. Should they? Or are they just placeholders for now?

Thank you for all the comments. When clicking on the `Edit` buttons, there should be an `<input>` element, and there should be `cancel` and `publish` buttons. I'll look into this to see if there is any bug here.

Also I will fix the query thing and check if I use the right type checking tools then.

File: web-app/django/VIM/apps/instruments/views/instrument_detail.py (hunk 4 above)
Comment (dchiller) [reply web-app/django/VIM/apps/instruments/views/instrument_detail.py:2]: I realize my message wasn't clear...I'll fix the typing dependencies, but would you remove the `Language` import?

Comment (kunfang98927) [reply web-app/django/VIM/apps/instruments/views/instrument_detail.py:2]: > I realize my message wasn't clear...I'll fix the typing dependencies, but would you remove the `Language` import?

Thank you for the clarification. In my latest commit, I kept the `Language` model for future support English and France detail page. The usage is similar to instrument list page:

```
active_language_en = self.request.session.get("active_language_en", None)
context["active_language"] = (
    Language.objects.get(en_label=active_language_en)
    if active_language_en
    else Language.objects.get(en_label="english")  # default in English
)
```

If it is necessary to support two languages, I'll keep the `Language` import.

Comment (dchiller) [reply web-app/django/VIM/apps/instruments/views/instrument_detail.py:2]: Ah ok, I didn't see that it is now used.

**Review by dchiller — APPROVED** [review]

**Review by yinanazhou — APPROVED** [review]

**Pull request merged**